frontend/src/Admin/components/EditShowForm.js

```diff
@@ -0,0 +1,262 @@
+            .required('Start Date is Required'),
+          entryEnd: yup
+            .date()
+            .nullable()
+            .isAfter(
+              yup.ref('entryStart'),
+              'Submission End Date must be after Submission Start Date'
+            )
+            .required('End Date is Required'),
+          judgingStart: yup
+            .date()
+            .nullable()
+            .isAfter(
+              yup.ref('entryEnd'),
+              'Judging Start Date must be after Submission End Date'
+            )
+            .required('Start Date is Required'),
+          judgingEnd: yup
+            .date()
+            .nullable()
+            .isAfter(
+              yup.ref('judgingStart'),
+              'Judging End Date must be after Judging Start Date'
+            )
+            .required('End Date is Required')
+        })}
+        onSubmit={values => {
+          values.entryStart
+          values.entryEnd
+          values.judgingStart
+          values.judgingEnd
+          update(values)
+            .then(() => done())
+            .catch(err => handleError(err.message))
+        }}
+        render={({
+          values,
+          errors,
+          touched,
+          setFieldValue,
+          setFieldTouched,
+          handleSubmit,
+          isSubmitting
+        }) => (
+          <Form onSubmit={handleSubmit}>
+            <Row>
+              <Col xs={6}>
+                <FormGroup>
+                  <Label>Name</Label>
+                  <Field
+                    type='text'
+                    id='name'
+                    name='name'
+                    className='form-control'
+                    required
+                  />
+                  {this.renderErrors(touched, errors, 'name')}
+                </FormGroup>
+                <FormGroup>
+                  <Label>Description</Label>
+                  <Field
+                    component='textarea'
+                    id='description'
+                    name='description'
+                    className='form-control'
+                    rows={3}
+                  />
+                  {this.renderErrors(touched, errors, 'description')}
+                </FormGroup>
+                <FormGroup>
+                  <Label>Individual Submission Limit</Label>
+                  <Field
+                    type='number'
+                    min='1'
+                    step='1'
+                    id='entryCap'
+                    name='entryCap'
+                    className='form-control'
+                    required
+                  />
+                  {this.renderErrors(touched, errors, 'entryCap')}
+                </FormGroup>
+              </Col>
+              <Col xs={6}>
+                <Row>
+                  <Col>
+                    <FormGroup>
+                      <Label>Submission Dates</Label>
+                      <CalendarContainer>
+                        <FormikDateRangePicker
```

Comment:
Is there a way to restrict this so that I can't set the start date to a day in the past? I could see that causing problems because it's a change that can't be undone. If we did that we would probably also want to set that restriction on creating shows (but that might make it hard to create test shows so I'm not too sure). Any thoughts anyone?

Reply:
I think that may not be super important, but I do remember Nannette had mentioned wanting restrictions on when a show's dates could be altered. I'm pretty sure it's just once the show entry period has started the entry start date can't change, and once the show has started judging the judging start date couldn't change.

Reply:
Nvm, I see that in there later.

Reply:
I've disabled the ability to set past dates  or present day when editing a show to avoid accidentally disabling edit functionality.

Reply:
Did we want the present day enabled?

Reply:
Selecting the present day locks it due to the start time being midnight.

Reply:
yeah locking present day could go either way. Might be something to bring up with Nanette, but I'll give it the okay for now.